salt/modules/zypper.py

```diff
@@ -58,65 +61,223 @@ def __virtual__():
     return __virtualname__
 
 
-def _zypper(*opts):
-    '''
-    Return zypper command with default options as a list.
-
-    opts
-        additional options for zypper command
-
-    '''
-    cmd = ['zypper', '--non-interactive']
-    cmd.extend(opts)
-
-    return cmd
-
-
-def _is_zypper_error(retcode):
-    '''
-    Return True in case the exist code indicate a zypper errror.
-    Otherwise False
-    '''
-    # see man zypper for existing exit codes
-    return int(retcode) not in [0, 100, 101, 102, 103]
+class _Zypper(object):
+    '''
+    Zypper parallel caller.
+    Validates the result and either raises an exception or reports an error.
+    Allows serial zypper calls (first came, first won).
+    '''
+
+    SUCCESS_EXIT_CODES = [0, 100, 101, 102, 103]
+    LOCK_EXIT_CODE = 7
+    XML_DIRECTIVES = ['-x', '--xmlout']
+    ZYPPER_LOCK = '/run/zypp.pid'
+    TAG_RELEASED = 'zypper/released'
+    TAG_BLOCKED = 'zypper/blocked'
+
+    def __init__(self, no_refresh=True):
+        '''
+        Constructor
+        :param no_refresh:
```

Comment:
Whoops! Forgot it after a massive refactoring. Removed. @cachedout, wasn't PyLint supposed to catch this?